### PR TITLE
Set username and password from Server URL if URL contains http basic auth

### PR DESCRIPTION
**Pull request by mnutt**
If user enters a server URL in the form `https://myuser:mypass@example.com`, on field blur, extract `myuser` and `mypass` and automatically fill them in the connection form.

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
Code seems fine to me, but why should an user type such an url and not use the three input fields as they are described? 
I mean why should be catch this improper use of the url field?

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 1]
```diff
@@ -790,12 +790,32 @@ private void onUrlInputFocusLost() {
...
+            Uri uri = Uri.parse(uriInput);
+
+            String userInfo = uri.getUserInfo();
+            if(userInfo != null) {
+                String username = userInfo.split(":", 2)[0];
+                if(username.length() != 0) {
+                    mUsernameInput.setText(username);
+                }
+                String password = userInfo.split(":", 2)[1];
+                if(password.length() != 0) {
```
Comment (przybylski) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: NPE for: http://username@example.com

Comment (przybylski) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Not NPE, but ArrayIndexOutOfBounds

**Comment by mnutt** [comment]
It's a bit easier for me to copy-and-paste a single URL from desktop to phone than to manually copy and paste the URL, username, and password separately, switching back and forth each time. It's not really meant to replace the three input fields, just to act as a shortcut for people who combine them.

It's also a part of the HTTP URI spec and other clients (web browsers, command line, etc) typically tend to support it.

**Comment by mnutt** [comment]
I fixed the ArrayIndexOutOfBounds by a slightly safer check for userInfo.

**Comment by przybylski** [comment]
Seems fine, I re-wrote extract function, let me know what do you think in terms of readability.

```java
String urlText = mHostUrlInput.getText();
if (TextUtils.isEmpty(urlText)) return;

String userInfo = Uri.parse(urlText.trim()).getUserInfo();

if (TextUtils.isEmpty(userInfo)) return;

String[] infoParts = userInfo.split(":", 2);

if (infoParts.length > 0 && !infoParts[0].isEmpty()) {
	mUsernameInput.setText(userInfoParts[0]);
	if (infoParts.length > 1 && !infoParts[1].isEmpty())
		mPasswordInput.setText(userInfoParts[1]);
}

```

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 2]
```diff
@@ -790,12 +790,31 @@ private void onUrlInputFocusLost() {
...
         } else {
             mOkButton.setEnabled(mServerIsValid);
             showRefreshButton(!mServerIsValid);
         }
     }
 
+    private void extractCredentialsFromUrl() {
+        String uriInput = mHostUrlInput.getText().toString().trim();
+
+        if(uriInput.length() != 0) {
```
Comment (przybylski) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: we hava a convention to put spaces after `if`s

**Comment by rperezb** [comment]
For tracking purposes: https://github.com/owncloud/ios/pull/515

**Comment by rperezb** [comment]
@mnutt it would be great if you may check whether the reported bugs on the iOS app works are not happening https://github.com/owncloud/ios/pull/515
thx!

**Pull request closed without merging**